src/Data/Collector.ts

```diff
@@ -432,6 +471,93 @@ export const verifyReceiptData = async (
   }
   // Determine the home partition index of the primary account (executionShardKey)
   const { homePartition } = ShardFunction.addressToPartition(cycleShardData.shardGlobals, executionShardKey)
+  if (globalModification) {
+    const appliedReceipt = receipt.signedReceipt as P2PTypes.GlobalAccountsTypes.GlobalTxReceipt
+    if (config.skipGlobalTxReceiptVerification) return { success: true }
+    else {
+      const { signs } = appliedReceipt
+      // Refer to https://github.com/shardeum/shardus-core/blob/7d8877b7e1a5b18140f898a64b932182d8a35298/src/p2p/GlobalAccounts.ts#L397
+      let votingGroupCount = cycleShardData.shardGlobals.nodesPerConsenusGroup
+      if (votingGroupCount > cycleShardData.nodes.length) {
+        if (nestedCountersInstance)
+          nestedCountersInstance.countEvent('receipt', 'votingGroupCount_greater_than_nodes_length')
+        Logger.mainLogger.error(
+          'votingGroupCount_greater_than_nodes_length',
+          votingGroupCount,
+          cycleShardData.nodes.length
+        )
+        // votingGroupCount = cycleShardData.nodes.length
+      }
+      let isReceiptMajority =
+        (signs.length / votingGroupCount) * 100 >= config.requiredMajorityVotesPercentage
+      if (!isReceiptMajority) {
+        Logger.mainLogger.error(
+          `Invalid receipt globalModification signs count is less than ${config.requiredMajorityVotesPercentage}% of the votingGroupCount, ${signs.length}, ${votingGroupCount}`
+        )
+        if (nestedCountersInstance)
+          nestedCountersInstance.countEvent(
+            'receipt',
+            `Invalid_receipt_globalModification_signs_count_less_than_${config.requiredMajorityVotesPercentage}%`
+          )
+        return result
+      }
+      
+      const nodeMap = new Map<string, P2PTypes.NodeListTypes.Node>()
```

Comment:
Since this map is only going to change once per cycle we can move this up the level for further optimisation. But for now LGTM.